src/services/programService/index.js

```diff
@@ -330,7 +333,12 @@ export const listDataCenters = async (shortName, jwt) => {
 		.then((response) => response.json())
 		.then((data) => {
 			if (data && Array.isArray(data)) {
-				return shortName ? getDataCenterByShortName(shortName, data) : data;
+				if (shortName) {
+					return getDataCenterByShortName(shortName, data);
+				} else if (id) {
+					return getDataCenterById(id, data);
+				}
+				return data;
```

Comment:
This function `listDataCenters` is confusing to me. 
Sometimes an `id` is passed, sometimes a `shortName` is passed and sometimes we just return the response without worrying about either.

Two suggested approaches:
1) change the function to take an options object. something like:
`const listDataCenters = async ({shortName, jwt, id}) => {..}`
and then there's no need to call this function with `undefined` parameters: https://github.com/icgc-argo/platform-api/pull/718/files#diff-36c4aa96f1bb25a05073048f02c5384b237fd2892af6c33d2d9dd7907d471e8eR154
then add some documentation with the `if/else` return section - it's not clear why we'd just return resp data 

2) my preferred method would be to extract the http request and break this into two functions that have a single responsibility. something like:
```js
const listDataCenters = async (jwt) => {
	const url = urljoin(PROGRAM_SERVICE_HTTP_ROOT, `/datacenters`);
	return await fetch(url, {
		method: 'get',
		headers: {
			Authorization: authorizationHeader(jwt),
		},
	})
		.then(restErrorResponseHandler)
		.then((response) => response.json())
};

export const getDataCenterByShortName = (shortName,jwt) => {
	const d = httpreq(jwt)
	return alreadyCreatedFilterFunc(d)
}

export const getDataCenterById = (id,jwt) => {
	const d = httpreq(jwt)
	return alreadyCreatedFilterFunc(d)
}


```
There's already the function logic in this file, it's just fully breaking it into it's own thing.
This way the functions are clear from just the name and are responsible for one thing.
You can reuse the common functionality of the HTTP req.

Reply:
Yes agree, this could have used a second look before opening PR for review, will probably go w/ exported functions, will review